src/sage/quadratic_forms/count_local_2.pyx

```diff
@@ -209,91 +198,87 @@ cdef local_solution_type_cdef(Q, p, w, zvec, nzvec):
     # (nzvec is non-empty and its components index a non-zero vector mod p)
-    if (nzvec is None):
+    if nzvec is None:
         nonzero_flag = True
-    elif (len(nzvec) == 0):
+    elif len(nzvec) == 0:
         nonzero_flag = False           # Trivially no solutions in this case!
     else:
         nonzero_flag = False
         i = 0
-        while ((not nonzero_flag) and (i < len(nzvec))):
-            if ((w[nzvec[i]] % p) != 0):
+        while not nonzero_flag and i < len(nzvec):
+            if w[nzvec[i]] % p:
                 nonzero_flag = True           # The non-zero condition is satisfied when we find one non-zero entry
             i += 1
 
     if not nonzero_flag:
         return <long> 0
 
-
     # Check if the solution has the appropriate (local) type:
     # -------------------------------------------------------
 
     # 1: Check Good-type
-    for i from 0 <= i < n:
-        if (((w[i] % p) != 0)  and ((Q[i,i] % p) != 0)):
+    for i in range(n):
+        if w[i] % p and Q[i, i] % p:
             return <long> 1
-    if (p == 2):
-        for i from 0 <= i < (n - 1):
-            if (((Q[i,i+1] % p) != 0) and (((w[i] % p) != 0) or ((w[i+1] % p) != 0))):
+    if p == 2:
+        for i in range(n - 1):
+            if Q[i, i+1] % p and (w[i] % p or w[i+1] % p):
                 return <long> 1
 
-
     # 2: Check Zero-type
     Zero_flag = True
-    for i from 0 <= i < n:
-        if ((w[i] % p) != 0):
+    for i in range(n):
+        if w[i] % p:
             Zero_flag = False
     if Zero_flag:
         return <long> 2
 
-
     # Check if wS1 is zero or not
     wS1_nonzero_flag = False
     for i from 0 <= i < n:
 
         # Compute the valuation of each index, allowing for off-diagonal terms
-        if (Q[i,i] == 0):
-            if (i == 0):
-                val = valuation(Q[i,i+1], p)    # Look at the term to the right
-            elif (i == n - 1):
-                val = valuation(Q[i-1,i], p)    # Look at the term above
+        if Q[i, i] == 0:
+            if i == 0:
+                val = valuation(Q[i, i+1], p)   # Look at the term to the right
+            elif i == n - 1:
+                val = valuation(Q[i-1, i], p)   # Look at the term above
             else:
-                val = valuation(Q[i,i+1] + Q[i-1,i], p)    # Finds the valuation of the off-diagonal term since only one isn't zero
+                val = valuation(Q[i, i+1] + Q[i-1, i], p)  # Finds the valuation of the off-diagonal term since only one isn't zero
         else:
-            val = valuation(Q[i,i], p)
+            val = valuation(Q[i, i], p)
 
         # Test each index
-        if ((val == 1) and ((w[i] % p) != 0)):
+        if val == 1 and w[i] % p:
             wS1_nonzero_flag = True
 
     # 4: Check Bad-type I
-    if (wS1_nonzero_flag is True):
+    if wS1_nonzero_flag:
         return <long> 4
 
     # 5: Check Bad-type II
-    if (wS1_nonzero_flag is False):
+    if wS1_nonzero_flag:
```

Comment:
This change looks dubious

Reply:
this is never assigned anything but boolean

```
git grep "wS1_non" src/
src/sage/quadratic_forms/count_local_2.pyx:    wS1_nonzero_flag = False
src/sage/quadratic_forms/count_local_2.pyx:            wS1_nonzero_flag = True
src/sage/quadratic_forms/count_local_2.pyx:    if (wS1_nonzero_flag is True):
src/sage/quadratic_forms/count_local_2.pyx:    if (wS1_nonzero_flag
``` is False):

Reply:
What I mean is: `wS1_nonzero_flag is False` is equivalent to `not wS1_nonzero_flag`, not `wS1_nonzero_flag`.

Reply:
oups, indeed